File: .claude/skills/dm/scripts/dm_formatter.py
#!/usr/bin/env python3
"""
dm_formatter.py — Formatta e salva DM freddi
Uso:
  dm_formatter.py --text "testo dm" --platform linkedin --lead "Mario Rossi"
  dm_formatter.py --text "testo dm" --platform x --lead "Mario Rossi" --copy
"""

import argparse
import sys
from datetime import date
from pathlib import Path

PLANS_DIR = Path(__file__).parents[4] / "plans"
WORD_LIMITS = {"linkedin": (150, 180), "x": (100, 140), "threads": (100, 130)}


def validate_length(text, platform):
    words = len(text.split())
    lo, hi = WORD_LIMITS.get(platform.lower(), (0, 9999))
    if words < lo:
        print(f"⚠️  DM troppo corto: {words} parole (minimo {lo} per {platform})")
    elif words > hi:
        print(f"⚠️  DM troppo lungo: {words} parole (massimo {hi} per {platform})")
    else:
        print(f"✓ Lunghezza OK: {words} parole ({lo}-{hi} per {platform})")
    return words


def save_dm(text, platform, lead_name):
    PLANS_DIR.mkdir(parents=True, exist_ok=True)
    safe_name = lead_name.lower().replace(" ", "_")
    filename = f"dm_{platform}_{safe_name}_{date.today().isoformat()}.txt"
    path = PLANS_DIR / filename
    path.write_text(text, encoding="utf-8")
    print(f"✓ DM salvato: {path}")
    return str(path)


def copy_to_clipboard(text):
    try:
        import pyperclip
        pyperclip.copy(text)
        print("✓ DM copiato negli appunti")
    except ImportError:
        print("⚠️  pyperclip non installato. Esegui: pip install pyperclip", file=sys.stderr)
    except Exception as e:
        print(f"⚠️  Impossibile copiare: {e}", file=sys.stderr)


def main(text, platform, lead_name, copy=False):
    print(f"\n--- DM {platform.upper()} — {lead_name} ---")
    print(text)
    print("---")
    validate_length(text, platform)
    save_dm(text, platform, lead_name)
    if copy:
        copy_to_clipboard(text)


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="DM Formatter")
    parser.add_argument("--text", required=True, help="Testo del DM")
    parser.add_argument("--platform", required=True, choices=["linkedin", "x", "threads"])
    parser.add_argument("--lead", required=True, help="Nome del lead")
    parser.add_argument("--copy", action="store_true", help="Copia negli appunti")
    args = parser.parse_args()
    main(args.text, args.platform, args.lead, copy=args.copy)

File: .claude/skills/dm/scripts/dm_formatter.js
#!/usr/bin/env node
/**
 * dm_formatter.js — Formatta e salva DM freddi multi-social
 * Uso:
 *   node dm_formatter.js --text "testo dm" --platform linkedin --lead "Mario Rossi"
 *   node dm_formatter.js --text "testo dm" --platform x --lead "Mario Rossi" --copy
 */

import { mkdirSync, writeFileSync } from 'fs';
import { resolve, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const PLANS_DIR = resolve(__dirname, '../../../../plans');

const WORD_LIMITS = { linkedin: [150, 180], x: [100, 140], threads: [100, 130] };

function validateLength(text, platform) {
  const words = text.trim().split(/\s+/).length;
  const [lo, hi] = WORD_LIMITS[platform.toLowerCase()] ?? [0, 9999];
  if (words < lo) console.log(`⚠️  DM troppo corto: ${words} parole (minimo ${lo} per ${platform})`);
  else if (words > hi) console.log(`⚠️  DM troppo lungo: ${words} parole (massimo ${hi} per ${platform})`);
  else console.log(`✓ Lunghezza OK: ${words} parole (${lo}-${hi} per ${platform})`);
  return words;
}

function saveDm(text, platform, leadName) {
  mkdirSync(PLANS_DIR, { recursive: true });
  const safeName = leadName.toLowerCase().replace(/\s+/g, '_');
  const today = new Date().toISOString().split('T')[0];
  const filename = `dm_${platform}_${safeName}_${today}.txt`;
  const path = resolve(PLANS_DIR, filename);
  writeFileSync(path, text, 'utf8');
  console.log(`✓ DM salvato: ${path}`);
  return path;
}

async function copyToClipboard(text) {
  try {
    const { default: clipboardy } = await import('clipboardy');
    clipboardy.writeSync(text);
    console.log('✓ DM copiato negli appunti');
  } catch {
    console.error('⚠️  Impossibile copiare negli appunti');
  }
}

async function main(text, platform, leadName, copy = false) {
  console.log(`\n--- DM ${platform.toUpperCase()} — ${leadName} ---`);
  console.log(text);
  console.log('---');
  validateLength(text, platform);
  saveDm(text, platform, leadName);
  if (copy) await copyToClipboard(text);
}

// CLI
const args = process.argv.slice(2);
const get = (flag) => { const i = args.indexOf(flag); return i !== -1 ? args[i + 1] : null; };

const text = get('--text');
const platform = get('--platform');
const lead = get('--lead');

if (!text || !platform || !lead) {
  console.log('Uso: node dm_formatter.js --text "testo" --platform linkedin|x|threads --lead "Nome" [--copy]');
  process.exit(1);
}

main(text, platform, lead, args.includes('--copy')).catch(e => { console.error(e.message); process.exit(1); });
